public/app/features/scenes/core/SceneDataProvider.tsx

```diff
@@ -0,0 +1,78 @@
+import { Unsubscribable } from 'rxjs/internal/types';
+
+import { TimeRange } from '@grafana/data';
+
+import { DataQueryExtended, SceneQueryRunner } from '../querying/SceneQueryRunner';
+
+import { SceneDataObject } from './SceneObjectBase';
+import { SceneDataState } from './types';
+
+export interface SceneDataProviderState extends SceneDataState {
+  queries: DataQueryExtended[];
+}
+
+export class SceneDataProvider extends SceneDataObject<SceneDataProviderState> {
+  private runner: SceneQueryRunner = new SceneQueryRunner({ queries: [] });
+  private querySub?: Unsubscribable;
+
+  constructor(state: SceneDataProviderState) {
+    super(state);
+    this.runner.queries = state.queries;
+  }
+
+  activate(): void {
+    super.activate();
+    const timeRange = this.getTimeRange();
+
+    this.subs.add(
+      timeRange.subscribe({
+        next: (range) => {
+          this.performQueries(range.range);
+        },
+      })
+    );
+
+    if (!this.state.$data) {
+      this.performQueries();
+    }
+  }
+
+  deactivate(): void {
+    super.deactivate();
+
+    if (this.querySub) {
+      this.querySub.unsubscribe();
+      this.querySub = undefined;
+    }
```

Comment:
I'd move this into an instance function to call it from `performQueries` as well. WDYT?